// Copyright The OpenTelemetry Authors
// SPDX-License-Identifier: Apache-2.0

syntax = "proto3";

option go_package = "flowmill.net/collector";

import "kubernetes_info.proto";

package collector;

message Info {
  enum Type {
    K8S_REPLICASET = 0;
    K8S_POD = 1;
  }

  Type type = 1;

  enum Event {
    ADDED = 0;
    MODIFIED = 1;
    DELETED = 2;
    ERROR = 3;
  }
  Event event = 2;

  PodInfo pod_info = 3;
  ReplicaSetInfo rs_info = 4;
}

message Response {
}

service Collector {
  rpc Collect(stream Info) returns (stream Response) {}
}
